### DISCUSSION TITLE
"Request scope" in Craft/Yii?

### DISCUSSION THREAD

**Discussion by wsydney76** [craftcms/cms#12951] [Q&A] [answered]
Still wondering what is the best way to use a "request scope" in Craft/Yii. Something that is described in a legacy system as "a variable scope that lives for the duration of a http/console request, and is accessible from everwhere", i.e. from every twig template, php code etc.

Useful for example to set a copyright notice in a header image template `request.copyright='Internet'` and outputting it in a footer template `&copy; #request.copyright#`. Or passing data from an `EVENT_BEFORE_SOMETHING` event handler to an `EVENT_AFTER_SOMETHING` handler.

Currently we are using this: 

```twig
{% do craft.app.urlManager.setRouteParams({copyright: image.copyright}, true) %}
---
{% set copyright = craft.app.urlManager.getRouteParams.copyright ?? '' %}
```

which works but looks semantically strange.

What do we miss?

/Mel

**Comment by brandonkelly** [comment]
Twig doesn’t have a built-in way of doing this. It’s come up a few times before, though, and there’s the [Transient](https://plugins.craftcms.com/transient?craft3) plugin, but it hasn’t yet been updated for Craft 4. So I just submitted a PR that would bring similar functionality into Craft natively: #12990

**Comment by brandonkelly** [comment, marked as answer]
Ended up taking a different approach. Craft 4.5 is going to introduce a new `_globals` global Twig variable, which is set to a collection, and be used to store globally-scoped values across all your templates. (#13050)

```twig
{% do _globals.set('copyright', image.copyright) %}
---
{{ _globals.get('copyright') }}
```

**Reply by brandonkelly** [reply]
Craft 4.5.0 is out with this! 🎉
